ext/opentelemetry-ext-otlp/src/opentelemetry/ext/otlp/metrics_exporter/__init__.py

```diff
@@ -0,0 +1,185 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""OTLP Metrics Exporter"""
+
+import logging
+
+from opentelemetry.ext.otlp.exporter import (
+    OTLPExporterMixin,
+    _get_resource_data,
+)
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2 import (
+    ExportMetricsServiceRequest,
+)
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2_grpc import (
+    MetricsServiceStub,
+)
+from opentelemetry.proto.common.v1.common_pb2 import StringKeyValue
+from opentelemetry.proto.metrics.v1.metrics_pb2 import (
+    DoubleDataPoint,
+    InstrumentationLibraryMetrics,
+    Int64DataPoint,
+)
+from opentelemetry.proto.metrics.v1.metrics_pb2 import (
+    Metric as CollectorMetric,
+)
+from opentelemetry.proto.metrics.v1.metrics_pb2 import (
+    MetricDescriptor,
+    ResourceMetrics,
+)
+from opentelemetry.sdk.metrics import (
+    Counter,
+    SumObserver,
+    UpDownCounter,
+    UpDownSumObserver,
+    ValueObserver,
+    ValueRecorder,
+)
+from opentelemetry.sdk.metrics.export import (
+    MetricsExporter,
+    MetricsExportResult,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _get_data_points(sdk_metric, data_point_class):
+
+    data_points = []
+
+    for (
+        label,
+        bound_counter,
+    ) in sdk_metric.instrument.bound_instruments.items():
+
+        string_key_values = []
+
+        for label_key, label_value in label:
+            string_key_values.append(
+                StringKeyValue(key=label_key, value=label_value)
+            )
+        data_points.append(
+            data_point_class(
+                labels=string_key_values,
+                value=bound_counter.aggregator.current,
+            )
+        )
+
+    return data_points
+
+
+def _get_temporality(instrument):
+    # pylint: disable=no-member
+    if isinstance(instrument, (Counter, UpDownCounter)):
+        temporality = MetricDescriptor.Temporality.DELTA
+    elif isinstance(instrument, (ValueRecorder, ValueObserver)):
+        temporality = MetricDescriptor.Temporality.INSTANTANEOUS
```

Comment:
FYI lots of changes incoming 😰 https://github.com/open-telemetry/opentelemetry-proto/pull/162